examples/advanced/pi-lluminate.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+
+# Use the Enviro pHAT to check how much light is in a room, then trigger a webhook to control
+# Philips Hue lights via IFTTT. By Wesley Archer (@raspberrycoulis).
+
+# Import the relevant modules
+
+import requests
+import time
+import datetime
+import os
+from envirophat import light
+
+# Get the current time for displaying in the terminal.
+def whats_the_time():
+    now = datetime.datetime.now()
+    return (now.strftime("%H:%M:%S"))
+
+# The function to turn off the lights. Sends a webhook to IFTTT which
+# triggers Hue. Replace the trigger word and tokens from your account.
+
+def turn_off():
+    requests.post("https://maker.ifttt.com/trigger/{TRIGGER_WORD}/with/key/{TOKEN_GOES_HERE}")
+    print("Lights off!")
+
+# The function to turn on the lights. Sends a webhook to IFTTT which                   
+# triggers Hue. Replace the trigger word and tokens from your account.
+
+def turn_on():
+    requests.post("https://maker.ifttt.com/trigger/{TRIGGER_WORD}/with/key/{TOKEN_GOES_HERE}")
+    print("Lights on!")
+
+# Check the light level and determine whether the lights need to 
+# be turned on or off.
+
+def average_lux():
+    # Variables for calculating the average lux levels
+    start_time = time.time()
+    curr_time = time.time()
+    collect_light_time = 60
+    collect_light_data = []
+
+    # Calculate the average lux level over 60 seconds
+    print("Calculating average light level...")
+    while curr_time - start_time < collect_light_time:
+	      curr_time = time.time()
+	      avg = light.light()
+	      collect_light_data.append(avg)
+	  time.sleep(1)
+    average_light = sum(collect_light_data[-10:]) / 10.0
+    now = whats_the_time()
+    print("{} {} {} {} {} {} {} {}.".format("Average over", collect_light_time, "seconds", "is:", average_light, "lux.", "Last checked at", now))
```

Comment:
Your format strings are a little contrived, you don't need to build the string from individual words/sentences, but can interpolate variables into a phrase. In this case you probably want:

```python
print("Average over {collect_time} seconds is: {average} lux. Last checked at {time}".format(
    collect_time=collect_light_time,
    average=average_light,
    time=now
))
```